chainntnfs/btcdnotify/btcd.go

```diff
@@ -223,6 +223,21 @@ func (b *BtcdNotifier) startNotifier() error {
 		return err
 	}
 
+	// Before we fetch the best block/blockheight we need to register the
+	// notifications for connected blocks, otherwise we might think we are
+	// at an earlier blockheight because during block notification we
+	// might have already mined some new blocks. Hence we will not get
+	// notified accordingly.
+	//
+	// NOTE: This is only a problem in itests where the block generation
+	// happens very fast and there is a chance of blocks being mined
+	// between two rpc calls.
+	if err := b.chainConn.NotifyBlocks(); err != nil {
```

Comment:
I think there's still this edge case, that if a new block is mined between calling `RegisterBlockEpochNtfn` in blockbeat dispatch and this method, we'd still have this issue, same for bitcoind.

To completely fix this issue, we need to make sure `NotifyBlocks` always notify the best block it knows atm - sth to fix in the future, maybe during the `btcwallet` refactor series.

Reply:
hmm I am not quite sure, because if a new block is found the "bestblock" is also increased, so during the `RegisterBlockEpochNtfn` the updated bestblock would be used 🤔